src/rust/include/rust/orchard/wallet.h

```diff
@@ -92,24 +92,32 @@ static_assert(
     "RawOrchardActionIVK struct should have exactly a 128-bit in-memory representation.");
 static_assert(alignof(RawOrchardActionIVK) == 8, "RawOrchardActionIVK struct alignment is not 64 bits.");
 
-typedef void (*push_action_ivk_callback_t)(void* resultCollection, const RawOrchardActionIVK actionIvk);
+typedef void (*push_action_ivk_callback_t)(void* rec, const RawOrchardActionIVK actionIvk);
+
+typedef void (*push_spend_action_idx_callback_t)(void* rec, uint32_t actionIdx);
 
 /**
  * Searches the provided bundle for notes that are visible to the specified wallet's
  * incoming viewing keys, and adds those notes to the wallet.  For each note decryptable
  * by one of the wallet's keys, this method will insert a `RawOrchardActionIVK` value into
- * the provided `resultCollection` using the `push_cb` callback. Note that this callback
- * can perform transformations on the provided RawOrchardActionIVK in this process.
+ * the provided `callbackReceiver` referent using the `push_cb` callback. Note that
+ * this callback can perform transformations on the provided RawOrchardActionIVK in this
+ * process.  For each action spending one of the wallet's notes, this method will insert
+ * a `uint32_t` action index into the `spendingActionIdxs` referent.
```

Comment:
```suggestion
 * process.  For each action spending one of the wallet's notes, this method will pass
 * a `uint32_t` action index corresponding to that action to the `callbackReceiver` referent; 
 * using the specified callback; usually, this will push the value into a result vector owned 
 * by the caller.
```